provider/server/machine_delete_test.go

```diff
@@ -80,7 +181,31 @@ var _ = Describe("DeleteMachine", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 
-		By("ensuring machine is deleted")
+		By("ensuring machine is in Terminating state after delete")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status.State
+		}).Should(Equal(iri.MachineState_MACHINE_TERMINATING))
+
+		By("ensuring machine is not deleted until gracefulShutdownTimeout")
+		Consistently(func() bool {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return err == nil && len(listResp.Machines) == 1
+		}, gracefulShutdownTimeout, consistentlyDuration).Should(BeTrue())
```

Comment:
Pls use methods for set consistently behavior: https://onsi.github.io/gomega/#consistently

consistentlyDuration is wrong name because it isn't duration but interval for polling.

Reply:
this isn't solve